RoaringBitmap/src/main/java/org/roaringbitmap/longlong/LongUtils.java

```diff
@@ -0,0 +1,82 @@
+package org.roaringbitmap.longlong;
+
+public class LongUtils {
+
+  //get the high 48 bit parts of the input data
+  public static byte[] highPart(long num) {
+    byte[] high48 = new byte[]{
+        (byte) ((num >>> 56) & 0xff),
+        (byte) ((num >>> 48) & 0xff),
+        (byte) ((num >>> 40) & 0xff),
+        (byte) ((num >>> 32) & 0xff),
+        (byte) ((num >>> 24) & 0xff),
+        (byte) ((num >>> 16) & 0xff)
+    };
+    return high48;
+  }
+
+  //get the low 16 bit parts of the input data
+  public static char lowPart(long num) {
+    return (char) num;
+  }
+
+  public static long toLong(byte[] high, char low) {
+    byte byte6 = (byte) (low >>> 8 & 0xff);
+    byte byte7 = (byte) low;
+    return (high[0] & 0xFFL) << 56
+        | (high[1] & 0xFFL) << 48
+        | (high[2] & 0xFFL) << 40
+        | (high[3] & 0xFFL) << 32
+        | (high[4] & 0xFFL) << 24
+        | (high[5] & 0xFFL) << 16
+        | (byte6 & 0xFFL) << 8
+        | (byte7 & 0xFFL);
+  }
+
+  //get little endian representation bytes
+  public static byte[] toLDBytes(long v) {
+    byte[] work = new byte[8];
+    work[0] = (byte) v;
+    work[1] = (byte) (v >> 8);
+    work[2] = (byte) (v >> 16);
+    work[3] = (byte) (v >> 24);
+    work[4] = (byte) (v >> 32);
+    work[5] = (byte) (v >> 40);
+    work[6] = (byte) (v >> 48);
+    work[7] = (byte) (v >> 56);
+    return work;
+  }
+
+  //get the long from the little endian representation bytes
+  public static long fromLDBytes(byte[] work) {
+    return (long) (work[7]) << 56 |
+        /* long cast needed or shift done modulo 32 */
+        (long) (work[6] & 0xff) << 48 | (long) (work[5] & 0xff) << 40
+        | (long) (work[4] & 0xff) << 32 | (long) (work[3] & 0xff) << 24
+        | (long) (work[2] & 0xff) << 16 | (long) (work[1] & 0xff) << 8 | (long) (work[0] & 0xff);
+  }
+
+  //compare according to the dictionary order
+  public static int compareHigh(byte[] a, byte[] b) {
+    return compareTo(a, 0, a.length, b, 0, b.length);
+  }
+
+  private static int compareTo(byte[] buffer1, int offset1, int length1,
+      byte[] buffer2, int offset2, int length2) {
+    if (buffer1 == buffer2 &&
```

Comment:
This should use `Arrays.mismatch` via shims unless it's only intended for very short arrays

Reply:
This comparator different from the mismatch meaning. It's to return the compare result.